src/main/java/org/apache/maven/plugins/dependency/ListClassesMojo.java

```diff
@@ -0,0 +1,326 @@
+                if ( !entryName.endsWith( ".class" ) )
+                {
+                    continue;
+                }
+
+                // remove .class from the end and change format to use periods instead of forward slashes
+                String className = entryName.substring( 0, entryName.length() - 6 ).replace( '/', '.' );
+                getLog().info( className );
+            }
+        }
+    }
+
+    private ProjectBuildingRequest makeBuildingRequest()
+            throws MojoExecutionException, MojoFailureException
+    {
+        if ( artifact == null )
+        {
+            throw new MojoFailureException( "You must specify an artifact, "
+                    + "e.g. -Dartifact=org.apache.maven.plugins:maven-downloader-plugin:1.0" );
+        }
+        if ( artifact != null )
+        {
+            String[] tokens = artifact.split( ":" );
+            if ( tokens.length < 3 || tokens.length > 5 )
+            {
+                throw new MojoFailureException( "Invalid artifact, you must specify "
+                        + "groupId:artifactId:version[:packaging[:classifier]] " + artifact );
+            }
+            coordinate.setGroupId( tokens[0] );
+            coordinate.setArtifactId( tokens[1] );
+            coordinate.setVersion( tokens[2] );
+            if ( tokens.length >= 4 )
+            {
+                coordinate.setType( tokens[3] );
+            }
+            if ( tokens.length == 5 )
+            {
+                coordinate.setClassifier( tokens[4] );
+            }
+        }
+
+        ArtifactRepositoryPolicy always =
+                new ArtifactRepositoryPolicy( true, ArtifactRepositoryPolicy.UPDATE_POLICY_ALWAYS,
+                        ArtifactRepositoryPolicy.CHECKSUM_POLICY_WARN );
+
+        List<ArtifactRepository> repoList = new ArrayList<>();
+
+        if ( pomRemoteRepositories != null )
+        {
+            repoList.addAll( pomRemoteRepositories );
+        }
+
+        if ( remoteRepositories != null )
+        {
+            // Use the same format as in the deploy plugin id::layout::url
+            String[] repos = remoteRepositories.split( "," );
+            for ( String repo : repos )
+            {
+                repoList.add( parseRepository( repo, always ) );
+            }
+        }
+
+        ProjectBuildingRequest buildingRequest =
+                new DefaultProjectBuildingRequest( session.getProjectBuildingRequest() );
+
+        Settings settings = session.getSettings();
+        repositorySystem.injectMirror( repoList, settings.getMirrors() );
+        repositorySystem.injectProxy( repoList, settings.getProxies() );
+        repositorySystem.injectAuthentication( repoList, settings.getServers() );
+
+        buildingRequest.setRemoteRepositories( repoList );
+
+        return buildingRequest;
+    }
+
+    protected ArtifactCoordinate toArtifactCoordinate( DependableCoordinate dependableCoordinate )
```

Comment:
why protected?

Reply:
Added via rebase of master. Should it be changed in this branch?